src/tracers/tracer.rs

```diff
@@ -1,11 +1,8 @@
-use crate::{eravm_error::EraVmError, execution::Execution, Opcode};
+use crate::{execution::Execution, state::VMState, Opcode};
 
 pub trait Tracer {
-    fn before_execution(
-        &mut self,
-        _opcode: &Opcode,
-        _vm: &mut Execution,
-    ) -> Result<(), EraVmError> {
-        Ok(())
-    }
+    fn before_decoding(&mut self, execution: &mut Execution, state: &mut VMState);
+    fn after_decoding(&mut self, opcode: &Opcode, execution: &mut Execution, state: &mut VMState);
+    fn before_execution(&mut self, opcode: &Opcode, execution: &mut Execution, state: &mut VMState);
+    fn after_execution(&mut self, opcode: &Opcode, execution: &mut Execution, state: &mut VMState);
```

Comment:
Why are we dropping `Result` here?

Reply:
Because tracers are functions to be accepted from the outside word. The vm executes them but doesn't care about what they do, thus they shouldn't stop the execution or make the vm panic because, what would be the reason?

Reply:
> The vm executes them but doesn't care about what they do

This seems correct.

>  Thus they shouldn't stop the execution or make the vm panic

This too, but note that code like [this](https://github.com/lambdaclass/era_vm/pull/212/files#diff-15d730b3f5d168f25f65469c1db5d90300d85cd8c479aad0988850978dd8a772R28):
```rs
let (src0, src1) = address_operands_read(vm, opcode).unwrap();
```

Will stop the execution of out vm due to a panic.

Reply:
Oh yes, good catch, I'll make sure to change that 😁